### ISSUE TITLE
XMP support

### ISSUE THREAD

**Issue by Beep6581** [Beep6581/RawTherapee#463]
Originally reported on Google Code with ID 473

```
To get someday XMP support to RT I put one idea where to start.

CMakeLists.txt add:

option (WITH_EXIV2 "Build with Exiv2 support" ON)
if (WITH_EXIV2)
    add_definitions (-DEXIV2)
    pkg_check_modules (EXIV2    REQUIRED exiv2>=0.19)
endif (WITH_EXIV2)

Examples can found: http://www.exiv2.org/
Exiv2 looks use Adobe XMP toolkit source for decoding XMP data.

```

Reported by `GreatBull69` on 2011-01-10 09:16:43
- **Blocking**: #1307, #1130

**Comment by asp** [comment]
Hi all, regarding where XMP fits into the image metadata universe, see the Metadata Working Group Guidelines:

    http://www.metadataworkinggroup.org/pdf/mwg_guidance.pdf

**Comment by ve3** [comment]
Yes, please make xmp support or at least can import xmp basic data like Darktable did.

**Comment by WestleyArgentum** [comment]
Hey, I'm trying to develop a better photo editing workflow so I'm trying out RawTherapee. So far I really like it, so thanks!

The one big issue for me is support for XMP sidecar files. I really like to import and star rank photos with tools like Photo Mechanic, but those tools all write the data out in ways that RawTherapee doesn't recognize (usually XMP sidecar files, from my experience so far). I noticed on the RawTherapee wikipedia page it says "... work is underway on migrating the PP3 sidecar system to XMP...". Is this true / is it a priority?

If so I may be interested in contributing (or at the very least testing), but I thought I'd ask first since this issue has been around for a long time.

Thanks!

**Comment by Beep6581** [comment]
@WestleyArgentum the work was abandoned. However, there has been a new effort, see branch `embedded-ratings`, or wait until PR #5325 is merged.

**Comment by JGKle** [comment]
I'm looking for an alternative to Lightroom, which I use for postprocessing JPG as well as RAW.  When working on JPG, the LR option _Catalog Settings->Metadata tab->Include Develop settings in metadata inside jpeg, tiff, and psd files_ causes it to directly embed all the edits into the original image's metadata, avoiding the need for a separate XMP file (while still being non-destructive on the pixels).

Although I'm pretty sure both RawTherapee & DarkTable lack such an option out of the box, I was hoping to use an external tool to embed the edits as a post-step (i.e. _exiftool -tagsfromfile_).  However, although I generally like RawTherapee much better than DarkTable...I realized that it uses a nonstandard/proprietary PP3 sidecar file, which seems to make this impossible.

That's what brought me to this thread. I see it was last discussed 8 months ago.  Is there still no hope of having its edits stored in XMP rather than PP3?  The last reply said to wait until #5325 is merged, which it now is.

(...Or alternatively, did I perhaps overlook some way to save the edits directly into the source jpg's metadata, similar to the LR option, which would effectively solve the same issue?)
